uniffi_docs/src/lib.rs

```diff
@@ -0,0 +1,518 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::{collections::HashMap, fs::read_to_string, path::Path, str::FromStr};
+
+use anyhow::Result;
+use pulldown_cmark::{Event, HeadingLevel::H1, Parser, Tag};
+use syn::Attribute;
+
+/// Function documentation.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Function {
+    pub description: String,
+    pub arguments_descriptions: HashMap<String, String>,
```

Comment:
I wonder if a `Vec<(String, String)>` or similar might be better so the args can be rendered in the order they are seen? Ditto the other hashmaps here, eg, an enum would probably want to render docs for an enum in the order the variants are declared?

Reply:
Rendered order is based on `arguments()` function of `uniffi_bindgen::interface::function::Function`:
https://github.com/eigerco/uniffi-rs/blob/3e04e2f60fe8f1af136110c2e1f3bdac800c34fe/uniffi_bindgen/src/bindings/kotlin/templates/FunctionDocsTemplate.kt#L8-L12